client/utils/should-link-be-shown.js

```diff
@@ -1,4 +1,21 @@
-const shouldLinkBeShown = (link, isAuthenticated) => {
+/* eslint-disable camelcase */
+const shouldLinkBeShown = (link, isAuthenticated, userData) => {
+  const {is_verified} = userData;
+  if (
+    link.authenticated === isAuthenticated &&
+    isAuthenticated === true &&
+    link.verified !== undefined &&
+    is_verified !== undefined
+  ) {
+    return link.verified === is_verified;
+  }
+  if (
+    link.authenticated === isAuthenticated &&
+    isAuthenticated === true &&
+    link.method !== undefined
+  ) {
+    return link.method === "mobile_phone";
```

Comment:
This is not working according to my test.
And indeed it looks quite wrong doesn't it? Why do you test if the link method is "mobile_phone"?
I think it should be: `return link.method === userData.method;`, this fixes the behavior in my manual testing, but please change your tests to catch this mistake and let me know which test is the failing one once it's done, so I will double check it.